### PR TITLE
add multipanel norm

### PR DESCRIPTION
**Pull request by rvitancol**
Need to bump the version if merging to dev.

### CODE REVIEW

File: R/NanoStringGeoMxSet-normalize.R [hunk 1]
```diff
@@ -14,68 +14,112 @@ HOUSEKEEPERS <- c('C1orf43','GPI','OAZ1','POLR2A','PSMB2','RAB7A',
...
+        stop ("Error: Negative Background normalization is for collapsed data set.
+        Run function aggregateCounts() to collapse the probes to targets.\n")
+    }
+    if (is.null(fData(object)[["Module"]]))
+    {
+        stop ("Error: Module is not specified in the object. Check your GeoMxSet object. \n")
+    }
+
+    # check if single panel
+    pools <- as.list(unique(fData(object)[["Module"]]))
```
Comment (NicoleEO) [inline R/NanoStringGeoMxSet-normalize.R:64]: I notice this is quite different from the plugin neg norm. That one should be optimized for time and I'm pretty sure it works with single panel as well. I don't think you need to split. I would just replace 65-112 with the neg norm plugin logic altogether. Let me know if you need help or if I am wrong and this doesn't work with single panel.

File: tests/testthat/test_normalization.R [hunk 2]
```diff
@@ -63,15 +63,59 @@ test_that("quantile norm values are correct", {
...
+test_that("negative norm values are correct for single panel", {
+  expect_equal(expectedOutputData, actualOutputData)
+})
+
+#### req 4a verify calculation of negative norm factors for multipanel
+# compute neg_norm factors for multipanel
+# call negative normalization
+target_demoData <- normalize(target_demoData , data_type="RNA", norm_method="neg",
+                                 toElt = "neg_norm")
+if (length(unique(fData(target_demoData)[["Module"]]) > 1)){
```
Comment (NicoleEO) [inline tests/testthat/test_normalization.R:90]: Remove if since we know it is >1 in the test

File: tests/testthat/test_normalization.R [hunk 3]
```diff
@@ -63,15 +63,59 @@ test_that("quantile norm values are correct", {
...
+  expectedOutputData <- (cbind("VnV_GeoMx_Hs_CTA_v1.2" = expectedOutputData_1,
+                               "Six-gene_test_v1_v1.1" = expectedOutputData_2))
+  actualOutputData <- pData(target_demoData)[["neg_norm_negFactors"]]
+}
+test_that("negative norm factors for multipanel are correct", {
+  expect_equal(expectedOutputData, actualOutputData)
+})
+
+# check the normalized values for multipanel
+if (length(unique(fData(target_demoData)[["Module"]]) > 1)){
```
Comment (NicoleEO) [inline tests/testthat/test_normalization.R:104]: remove if here too

**Review by NicoleEO — CHANGES_REQUESTED** [review]

File: R/NanoStringGeoMxSet-normalize.R (hunk 1 above)
Comment (rvitancol) [reply R/NanoStringGeoMxSet-normalize.R]: @NicoleEO Please check. I updated these. It seems that the lines moved so there are a lot of changes. But I only updated lines 65 to 112

File: tests/testthat/test_normalization.R (hunk 2 above)
Comment (rvitancol) [reply tests/testthat/test_normalization.R]: done

File: tests/testthat/test_normalization.R (hunk 3 above)
Comment (rvitancol) [reply tests/testthat/test_normalization.R:103]: done

**Pull request merged**